### PR TITLE
PIMS-420: Routes - Buildings and restructuring interfaces

### PR DESCRIPTION
**Pull request by LawrenceLau2020**
## 🎯 Summary

  
[PIMS-420: ](https://citz-imb.atlassian.net/browse/PIMS-420) - PIMS-420: Routes - Buildings and restructuring interfaces

Created 2 base classes IProperty and IBaseEntity.
Created route and controller for buildings and also breaking out some common fields for the IProperty and the IBaseEntity interfaces that will be inherited from other interfaces.

## 🔰 Checklist

- [x] I have read and agree with the following checklist and am following the guidelines in our [Code of Conduct](CODE_OF_CONDUCT.md) document.

> - I have performed a self-review of my code.
> - I have commented my code, particularly in hard-to-understand areas.
> - I have made corresponding changes to the documentation where required.
> - I have tested my changes to the best of my ability.
> - My changes generate no new warnings.

### CODE REVIEW

File: express-api/src/controllers/baseEntity.ts [hunk 1]
```diff
@@ -0,0 +1,8 @@
+import { UUID } from 'crypto';
+
+export interface IBaseEntity {
+  createdOn: string;
+  updatedOn: string;
+  createdById: UUID;
+  updatedById: UUID;
+}
```
Comment (LawrenceLau2020) [inline express-api/src/controllers/baseEntity.ts:8]: I'm not sure if this is the best place for the baseEntity.ts interface which is the root level of the "controllers" folder.

Comment (dbarkowsky) [reply express-api/src/controllers/baseEntity.ts:8]: Maybe we could have a `common` folder? For things that apply generally to many controllers.

Comment (LawrenceLau2020) [reply express-api/src/controllers/baseEntity.ts]: Makes sense.

File: express-api/src/controllers/common/IBaseEntity.ts [hunk 2]
```diff
@@ -0,0 +1,8 @@
+import { UUID } from 'crypto';
+
+export interface IBaseEntity {
+  createdOn: string;
+  updatedOn: string;
+  createdById: UUID;
+  updatedById: UUID;
+}
```
Comment (dbarkowsky) [inline express-api/src/controllers/common/IBaseEntity.ts:8]: I think that the updated fields might be undefined at times if something had been created but never updated. They are also nullable in the database.
```suggestion
export interface IBaseEntity {
  createdOn: string;
  updatedOn?: string;
  createdById: UUID;
  updatedById?: UUID;
}
```

File: express-api/src/routes/buildingsRouter.ts [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+export const BUILDINGS_ROUTE = '/properties/buildings';
+
+const router = express.Router();
+
+// Endpoints for buildings data manipulation
+router.route(`${BUILDINGS_ROUTE}/:buildingId`).get(controllers.getBuilding);
+router.route(`${BUILDINGS_ROUTE}/:buildingId`).put(controllers.updateBuilding);
+router.route(`${BUILDINGS_ROUTE}/:buildingId`).delete(controllers.deleteBuilding);
+router.route(BUILDINGS_ROUTE).get(controllers.filterBuildingsQueryString);
+router.route(BUILDINGS_ROUTE).post(controllers.addBuilding);
```
Comment (dbarkowsky) [inline express-api/src/routes/buildingsRouter.ts:13]: You can actually chain together http method calls if they use the same route:
```suggestion
router.route(`${BUILDINGS_ROUTE}/:buildingId`)
  .get(controllers.getBuilding)
  .put(controllers.updateBuilding)
  .delete(controllers.deleteBuilding);
router.route(BUILDINGS_ROUTE)
  .get(controllers.filterBuildingsQueryString)
  .post(controllers.addBuilding);
```

**Review by dbarkowsky — APPROVED** [review]

**Pull request merged**